packages/pyodide-kernel/src/worker.ts

```diff
@@ -80,6 +85,28 @@ export class PyodideRemoteKernel {
       await micropip.install('${pipliteWheelUrl}', keep_going=True)
     `);
     }
+    console.debug('extraPackagesAndIndexes', extraPackagesAndIndexes);
+
+    if (extraPackagesAndIndexes.length > 0) {
+      // note that here pkg can be a package name or a wheel url
+      for (let { package: pkg, indexes } of extraPackagesAndIndexes) {
+        let installCmd: string;
+        if (indexes === null) {
+          installCmd = `import micropip\nawait micropip.install('${pkg}', keep_going=True)`;
+        } else {
+          installCmd = `import micropip\nawait micropip.install('${pkg}', index_urls=${JSON.stringify(indexes)}, keep_going=True)`;
+        }
+        console.info('installCmd', installCmd);
+        try {
+          await this._pyodide.runPythonAsync(installCmd);
+          console.info(`Package ${pkg} Installed successfully`);
+        } catch (e) {
+          console.error('Error installing package', e);
+        }
+      }
+    } else {
+      console.info('no extra packages and indexes');
+    }
 
     // get piplite early enough to impact pyodide-kernel dependencies
     await this._pyodide.runPythonAsync(`
```

Comment:
I think lower in initKernel the 

```
    for (const pkgName of toLoad) {
      if (!preloaded.includes(pkgName)) {
        scriptLines.push(`await piplite.install('${pkgName}', keep_going=True)`);
      }
    }
```

Could be modified to be a single call to `piplite.install(list)`, instead of repeated install calls.